apps/heft/src/plugins/TypeScriptPlugin/TypeScriptBuilder.ts

```diff
@@ -885,8 +887,45 @@ export class TypeScriptBuilder extends SubprocessRunnerBase<ITypeScriptBuilderCo
       outFolderPath = path.resolve(this._configuration.buildFolder, outFolderPath);
     }
 
+    outFolderPath = Path.convertToSlashes(outFolderPath);
+    outFolderPath = outFolderPath.replace(/(\/*)$/, '/'); // Ensure the outFolderPath ends with a slash
+
+    for (const existingModuleKindToEmit of this._moduleKindsToEmit) {
+      let errorText: string | undefined;
+
+      if (existingModuleKindToEmit.outFolderPath === outFolderPath) {
+        if (existingModuleKindToEmit.jsExtensionOverride === jsExtensionOverride) {
+          errorText =
+            'Unable to output two different module kinds with the same ' +
+            `module extension (${jsExtensionOverride || '.js'}) to the same ` +
+            `folder ("${outFolderPath}").`;
+        }
+      } else {
+        let parentFolder: string | undefined;
+        let childFolder: string | undefined;
+        if (outFolderPath.startsWith(existingModuleKindToEmit.outFolderPath)) {
```

Comment:
Technically we can, as long as the extensions are different there can't be a collision, but I guess this might cause issues with static copy?

Reply:
It apparently causes problems in watch mode.